File: pdu/python/hako_srv_msgs/pdu_cdr_conv_JoinRequestPacket.py
from .pdu_pytype_JoinRequestPacket import JoinRequestPacket
from ..pdu_cdr_runtime import CdrReader, CdrWriter

from ..hako_srv_msgs.pdu_cdr_conv_JoinRequest import *
from ..hako_srv_msgs.pdu_cdr_conv_ServiceRequestHeader import *



def py_to_cdr_body_JoinRequestPacket(writer: CdrWriter, src: JoinRequestPacket):
    py_to_cdr_body_ServiceRequestHeader(writer, src.header)
    py_to_cdr_body_JoinRequest(writer, src.body)


def cdr_body_to_py_JoinRequestPacket(reader: CdrReader, dst: JoinRequestPacket):
    cdr_body_to_py_ServiceRequestHeader(reader, dst.header)
    cdr_body_to_py_JoinRequest(reader, dst.body)
    return dst


def py_to_cdr_JoinRequestPacket(src: JoinRequestPacket) -> bytes:
    writer = CdrWriter()
    writer.write_encapsulation()
    py_to_cdr_body_JoinRequestPacket(writer, src)
    return writer.bytes()


def cdr_to_py_JoinRequestPacket(cdr_payload) -> JoinRequestPacket:
    reader = CdrReader(cdr_payload)
    reader.read_encapsulation()
    dst = JoinRequestPacket()
    return cdr_body_to_py_JoinRequestPacket(reader, dst)

File: pdu/javascript/hako_srv_msgs/pdu_cdr_conv_JoinRequestPacket.js
import { JoinRequestPacket } from './pdu_jstype_JoinRequestPacket.js';
import { PduCdrWriter, PduCdrReader } from '../pdu_cdr_runtime.js';
import { JoinRequest } from '../hako_srv_msgs/pdu_jstype_JoinRequest.js';
import { PduJoinRequestConverter } from '../hako_srv_msgs/pdu_cdr_conv_JoinRequest.js';
import { ServiceRequestHeader } from '../hako_srv_msgs/pdu_jstype_ServiceRequestHeader.js';
import { PduServiceRequestHeaderConverter } from '../hako_srv_msgs/pdu_cdr_conv_ServiceRequestHeader.js';


export class PduJoinRequestPacketConverter {
    /**
     * @param {PduCdrWriter} writer
     * @param { JoinRequestPacket } src
     */
    static to_cdr_body(writer, src) {
        PduServiceRequestHeaderConverter.to_cdr_body(writer, src.header);
        PduJoinRequestConverter.to_cdr_body(writer, src.body);
    }

    /**
     * @param {PduCdrReader} reader
     * @param { JoinRequestPacket } dst
     * @returns { JoinRequestPacket }
     */
    static cdr_body_to_js(reader, dst) {
        PduServiceRequestHeaderConverter.cdr_body_to_js(reader, dst.header);
        PduJoinRequestConverter.cdr_body_to_js(reader, dst.body);
        return dst;
    }

    /**
     * @param { JoinRequestPacket } src
     * @returns {ArrayBuffer}
     */
    static to_cdr(src) {
        const writer = new PduCdrWriter();
        writer.write_encapsulation();
        this.to_cdr_body(writer, src);
        return writer.get_buf();
    }

    /**
     * @param {ArrayBuffer|ArrayBufferView} cdrPayload
     * @returns { JoinRequestPacket }
     */
    static from_cdr(cdrPayload) {
        const reader = new PduCdrReader(cdrPayload);
        reader.read_encapsulation();
        return this.cdr_body_to_js(reader, new JoinRequestPacket());
    }
}
